include/f_core/flight/c_phase_controller.h

```diff
@@ -0,0 +1,207 @@
+#ifndef F_CORE_FLIGHT_C_PHASE_CONTROLLER_H
+#define F_CORE_FLIGHT_C_PHASE_CONTROLLER_H
+
+#include <array>
+#include <cstdint>
+#include <cstdio>
+#include <zephyr/fs/fs.h>
+#include <zephyr/kernel.h>
+
+int flight_log_init(const char *flight_log_file_name, fs_file_t *fp);
+int flight_log_source_event(bool enabled, fs_file_t *fp, const char *source, const char *event);
+int flight_log_event_confirmed(bool enabled, fs_file_t *fp, const char *event, bool currentState);
+int flight_log_print(bool enabled, fs_file_t *fp, const char *str);
+int flight_log_close(bool enabled, fs_file_t *fp);
```

Comment:
Can these just be declared in a separate header and be namespaced?

Reply:
yes. now its a class an optinalness is portrayed by a nullptr